### DISCUSSION TITLE
Does MuLan require attention masking for the audio part of the tower?

### DISCUSSION THREAD

**Discussion by 14leeyuchieh** [lucidrains/musiclm-pytorch#65] [Q&A]
I was taking a look at the `forward` implementation of `AudioSpectrogramTransformer` and notice it does not intake an attention mask or pass it to the underlying transformer

```
def forward(
        self,
        x,
        force_no_patch_dropout = False,
        return_all_layers = False
    ):
```

Is this a bug? Or does Mulan handle attention masking when batching audio of arbitrary length via just padding?
